File: src/main/java/com/itmo/microservices/tasks/controller/TaskController.java
package com.itmo.microservices.tasks.controller;

import com.itmo.microservices.tasks.entity.Task;
import lombok.AllArgsConstructor;
import org.springframework.web.bind.annotation.*;
import com.itmo.microservices.tasks.model.TaskDto;
import com.itmo.microservices.tasks.service.TaskService;

import java.util.List;
import java.util.UUID;

@RestController
@RequestMapping("/tasks")
@AllArgsConstructor
public class TaskController {

    private final TaskService taskService;

    @GetMapping
    public List<Task> getAllTasks() {
        return taskService.getAllTasks();
    }

    @GetMapping("/{taskId}")
    public Task getTaskById(@PathVariable UUID taskId) {
        return taskService.getTaskById(taskId);
    }

    @PostMapping
    public void addTask(@RequestBody TaskDto task) {
        taskService.addTask(task);
    }

    @PostMapping("/{taskId}/assignee/{username}")
    public void assignTask(@PathVariable UUID taskId, @PathVariable String username) {
        taskService.assignTask(taskId, username);
    }

    @DeleteMapping("/{taskId}")
    public void deleteTaskById(@PathVariable UUID taskId) {
        taskService.deleteTaskById(taskId);
    }
}

File: src/main/kotlin/com/itmo/microservices/demo/tasks/controller/TaskController.kt
package com.itmo.microservices.demo.tasks.controller

import com.itmo.microservices.demo.tasks.entity.Task
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.bind.annotation.RequestMapping
import lombok.AllArgsConstructor
import com.itmo.microservices.demo.tasks.service.TaskService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import java.util.UUID
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import com.itmo.microservices.demo.tasks.model.TaskDto
import org.springframework.web.bind.annotation.DeleteMapping

@RestController
@RequestMapping("/tasks")
@AllArgsConstructor
class TaskController(private val taskService: TaskService) {

    @GetMapping
    fun allTasks(): List<Task> = taskService.allTasks()

    @GetMapping("/{taskId}")
    fun getTaskById(@PathVariable taskId: UUID): Task = taskService.getTaskById(taskId)

    @PostMapping
    fun addTask(@RequestBody task: TaskDto) = taskService.addTask(task)

    @PostMapping("/{taskId}/assignee/{username}")
    fun assignTask(@PathVariable taskId: UUID, @PathVariable username: String) =
            taskService.assignTask(taskId, username)

    @DeleteMapping("/{taskId}")
    fun deleteTaskById(@PathVariable taskId: UUID) = taskService.deleteTaskById(taskId)
}